plugins/export_spec_3d_csv.py

```diff
@@ -0,0 +1,125 @@
+class SpecCSVPlugin(Plugin):
+    name = "Export Spectrum to 3D CSV"
+    __version__ = "1.0"
+    __author__ = u"Éric Piel"
+    __license__ = "Public domain"
+
+    def __init__(self, microscope, main_app):
+        super(SpecCSVPlugin, self).__init__(microscope, main_app)
+        self.addMenu("File/Export spectrum to 3D CSV...", self.export)
+
+    def export(self):
+        analysis_tab = self.main_app.main_data.getTabByName('analysis')
+
+        # Search for a spectrum stream
+        for s in analysis_tab.tab_data_model.streams.value:
+            if isinstance(s, StaticSpectrumStream):
+                specs = s
+                break
+        else:
+            box = wx.MessageDialog(self.main_app.main_frame,
+                       "No spectrum stream found to export.",
+                       "Failed to find spectrum stream", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # Select a file name
+        path = get_acqui_conf().last_export_path
+        wildcards, uformats = formats_to_wildcards({"CSV": [u"csv"]})
+        dialog = wx.FileDialog(self.main_app.main_frame,
+                           message="Choose a filename and destination",
+                           defaultDir=path,
+                           defaultFile="",
+                           style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+                           wildcard=wildcards)
+
+        # Show the dialog and check whether is was accepted or cancelled
+        if dialog.ShowModal() != wx.ID_OK:
+            return
+
+        # Get the full filename, and add an extension if needed
+        path = dialog.GetDirectory()
+        fn = dialog.GetFilename()
+        if not fn.lower().endswith(".csv"):
+            fn += ".csv"
+
+        fullfn = os.path.join(path, fn)
+        
+        # Export
+        self.export_3d_csv(fullfn, specs.calibrated.value)
+
+    @staticmethod
+    def export_3d_csv(filename, da):
+        """
+        filename (str)
+        da (DataArray of shape C11YX)
+        """
+        logging.debug("Will export data of shape %s to %s", da.shape, filename)
+        
+        # Get the metadata
+        spectrum_range, unit = spectrum.get_spectrum_range(da)
+        pxs = da.metadata.get(model.MD_PIXEL_SIZE, (1, 1))
+        
+        # Remove useless dimensions -> CYX
+        da = da[:, 0, 0, :, :]
+    
+        with open(filename, 'w') as fd:
+            csv_writer = csv.writer(fd)
+            csv_writer.writerow(["wavelength (%s)" % unit, "X (m)", "Y (m)", "intensity"])
+
+            l = 0
+            # Y
+            for iy in range(da.shape[-2]):
+                # X
+                for ix in range(da.shape[-1]):
+                    # Wavelength
+                    for wl, data in zip(spectrum_range, da[:, iy, ix]):
+                        csv_writer.writerow([wl, ix * pxs[0], iy * pxs[1], data])
+                        l += 1
+
+        logging.debug("Exported %d rows", l)
```

Comment:
Isn't `l` equal to `da.shape[-2] * da.shape[-1] * len(spectrum_range)`? It looks strange that you have to do the `l+=1` to get the number of rows exported.

Reply:
Indeed, it *should* be prod(da.shape). During development I wanted to check that it's indeed the case. 
This code is not needed anymore, so I'll remove it.